frappe/model/base_document.py

```diff
@@ -721,6 +721,8 @@ def db_insert(self, ignore_if_duplicate=False):
 		)
 
 		columns = list(d)
+		savepoint = "before_insert"
+		frappe.db.savepoint(savepoint)
```

Comment:
1. This is quite a lot of overhead TBH. If this is the only way, then limit it to postgres, why should mariadb users pay the price?
2. Release the savepoint if insertion was successful? Not sure what's more efficient here.


Alternatively... ANY creative solution that doesn't require savepoint?